go.mod

```diff
@@ -1,11 +1,21 @@
 module github.com/gametimesf/testy
 
-go 1.17
+go 1.18
```

Comment:
if go's release cadence maintains the same, this should be fully released this month. I am expecting on needing generics to implement some of the stuff in a less painful way, so figured I'd just bite that bullet now

Reply:
and now generics are in use